lara-typescript/src/interactive-api/mock-iframe-phone.ts

```diff
@@ -0,0 +1,238 @@
+export const realIframePhone = jest.requireActual("iframe-phone");
+
+class Messages {
+  private _messages: any[] = [];
+
+  public all() {
+    return this._messages;
+  }
+
+  public at(idx: number) {
+    return this._messages[idx];
+  }
+
+  public count() {
+    return this._messages.length;
+  }
+
+  public reset() {
+    this._messages = [];
+  }
+
+  public findType(type: string) {
+    // tslint:disable-next-line:prefer-for-of
+    for (let i = 0; i < this._messages.length; i++) {
+      if (this._messages[i].message.type === type) return this._messages[i];
+    }
+    return null;
+  }
+
+  public _add(msg: any) {
+    this._messages.push(msg);
+  }
+}
+
+let autoConnect = true;
+export const setAutoConnect = (value: boolean) => autoConnect = value;
+
+export class MockIframePhoneManager {
```

Comment:
Seems like this ought to be moved to the iframe-phone module, so it could be shared by other iframe-phone users.